File: 프로그래머스/lv1/이재표/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int[]arr=new int[number+1];
        for(int i=1;i<=number;i++){
            int mul=1;
            while(mul*i<=number){
                arr[i*mul]+=1;
                mul++;
            }
        }
        
        int answer = 0;
        for(int i=1;i<=number;i++){
            if(arr[i]<=limit){
                answer+=arr[i];
            }else{
                answer+=power;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/lv1/leehandsub/기사단원의 무기.kt
import java.lang.Math.sqrt

class Solution {
    fun solution(number: Int, limit: Int, power: Int): Int {
        var answer: Int = 0
        val arr = Array(number + 1) { 0 }
        for (i in 1 until number + 1) {
            for (j in 1 until sqrt(i.toDouble()).toInt() + 1) {
                if (i % j == 0) {
                    if (i == j * j)
                        arr[i] += 1
                    else
                        arr[i] += 2
                }
            }
        }
        for (i in arr) {
            answer += if (i > limit)
                power
            else
                i
        }
        return answer
    }
}
